src/tableformat.py

```diff
@@ -0,0 +1,112 @@
+from typing import List, Tuple
+
+
+class TableFormatter:
```

Comment:
I checked the exercises page and I am a little bit disappointed they don't mentioned the abstract classes!

Here, the problem is that `TableFormatter` is useless as is its (no implementation of `headings` and `rows`) but you can create instances which is counter-intuitive.

In order to prevent a user to create instances you can define the class as abstract. In order to do this you have 2 things to add:
- use `ABC` as base class
- use `@abstractmethod` decorator on the methods not implemented
```python
from abc import ABC, abstractmethod

class TableFormatter(ABC):
    
    @abstractmethod
    def headings(self, headers: Tuple[str, str, str, str]) -> List[str]:
        pass
    
    @abstractmethod
    def rows(self, rowdata: List[Tuple[str, int, float, float]]) -> List[str]:
        pass
```

And then if you try to create an instance you will have the following error message:
```
TableFormatter()
TypeError: Can't instantiate abstract class TableFormatter with abstract methods headings, rows
```

Without this you will have an error only after creating an instance and call one of the method not implemented. In general, it is always better to raise an error as soon as possible!

Reply:
Changed as per comment below.